src/App.js

```diff
@@ -1,16 +1,41 @@
 import React from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import { useState } from 'react'
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [chatData, setChatData] = useState(chatMessages)
+
+  const likeMessage = (id) => {
+    setChatData(chatData => chatData.map(chat => {
+      if(chat.id === id) {
+        return {...chat, liked: !chat.liked}
+      } else {
+        return chat
+      }
+    }))
+  }
+
+  const countLikedMessages = (chatData) => {
+    return chatData.reduce((total, chat) => {
+      return total + (chat.liked ? 1 : 0);
```

Comment:
Works perfectly! If you'd like to avoid an unnecessary calculation (i.e. adding by 0), you could use the ternary a bit differently:
```javascript
return chat.liked ? total + 1 : total;
```